generator/MarshalGen.cs

```diff
@@ -60,5 +60,22 @@ public override string FromNative (string var)
 			return String.Format (from_fmt, var);
 		}
 	}
+
+	public class StringMarshalGen : MarshalGen, IManualMarshaler
+	{
+		public StringMarshalGen (string ctype, string type, string mtype, string call_fmt, string from_fmt) : base (ctype, type, mtype, call_fmt, from_fmt, true)
+		{
+		}
+
+		public string AllocNative (string managed_var)
+		{
+			return CallByName (managed_var);
+		}
+
+		public string ReleaseNative (string native_var)
+		{
+			return $"GLib.Marshaller.Free ({native_var})";
```

Comment:
Jenkins does have Mono 4.4. I think we should bump the required version in the configure file.

In any case, I don't like this patch, but there's not much without rewriting the generator typesystem, and I prefer to do that in master.